xbmc/addons/AddonInstaller.cpp

```diff
@@ -395,12 +389,12 @@ bool CAddonInstaller::CheckDependencies(const AddonPtr &addon,
         (!haveInstalledAddon && !optional))
     {
       // we have it but our version isn't good enough, or we don't have it and we need it
-      if (!database.GetAddon(addonID, dep) || !dep->MeetsVersion(versionMin, version))
+      if (!dep->MeetsVersion(versionMin, version) ||
```

Comment:
@ksooo also a potential crash candidate. dep could be nullptr (if !haveInstalledAddon && !optional).
will pr later if you can‘t get to it